### PR TITLE
Bug 1551863 - Pass missing disabled flag to Rust side

### PR DESCRIPTION
**Pull request by badboy**

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt [hunk 1]
```diff
@@ -27,7 +27,8 @@ class BooleanMetricType(
                 name = name,
                 send_in_pings = ffiPingsList,
                 send_in_pings_len = sendInPings.size,
-                lifetime = lifetime.ordinal)
+                lifetime = lifetime.ordinal,
+                disabled = if (disabled) { 1 } else { 0 })
```
Comment (Dexterp37) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt]: What if we write an extension function for this, in `LibGleanFFI.kt`? We could do something like:

```
// Comment about what this does and why we need it
internal fun Boolean.toInt() = if (this) 1 else 0
```

This way the code above can be reduced to:

`disabled = disabled.toInt()`

File: glean-core/android/src/main/java/mozilla/telemetry/glean/private/CounterMetricType.kt [hunk 2]
```diff
@@ -43,7 +43,8 @@ class CounterMetricType(
                 name = name,
                 send_in_pings = ffiPingsList,
                 send_in_pings_len = sendInPings.size,
-                lifetime = lifetime.ordinal)
+                lifetime = lifetime.ordinal,
+                disabled = if (disabled) { 1 } else { 0 })
```
Comment (Dexterp37) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/private/CounterMetricType.kt]: nit: same consideration as above, here and in all the other places below.

**Comment by Dexterp37** [comment]
@badboy do you want to take enabling back the tests to a separate PR/bug?

**Comment by badboy** [comment]
> @badboy do you want to take enabling back the tests to a separate PR/bug?

Nope, let me actually re-enable them here

**Pull request merged**